resources/analysisTools/roddyTools/wrapInScript.sh

```diff
@@ -220,10 +220,14 @@ childProcesses() {
     # Note that the terminal sed in the following expression is necessary, because pstree seems to behave differently in interactive and non-
     # interactive mode. In non-interactive mode, pstree appends a '...' to every non-terminal process ID.
     declare -a pidList=( $(pstree -a -p $$ | cut -d, -f2 | cut -d" " -f1 | grep -v $$ | sed -r 's/\.*//g') )
+
+    ## To get a clean list of subprocesses we remove the PIDs of the cut, grep, and sed commands and that of the current subshell.
+    local currentSubshell=$(bash -c 'echo $PPID')  ## https://stackoverflow.com/questions/20725925/get-pid-of-current-subshell
     for pid in "${pidList[@]}"; do
-        # Remove the PIDs for (at least) the cat and grep commands.
-        if processesExist "$pid"; then
-            echo "$pid"
+        if [[ "$pid" != "$currentSubshellPid" ]]; then
+            if processesExist "$pid"; then
```

Comment:
Please put in [[ ]]

Reply:
Nope, this is neither necessary nor useful here. The processesExist function works like this. The alternative (which would be o.k. for me) would be to return something on STDOUT and test that.

Reply:
The main reason I did it like this, is that the actually called program in `processesExist` also just returns != if a process does not exist. Changing this to return a string would not only make code here more complex `[[ $(...) != "yes" ]]` but also there (`if $!; then ... fi`). Just something to consider.